ocs_ci/ocs/cluster_load.py

```diff
@@ -0,0 +1,278 @@
+
+        pvc_size = int(get_osd_pods_memory_sum() * 0.75)
+
+        self.logger.info(
+            f"In order to eliminate the OSD pods cache effect, creating 1 pod with a "
+            f"PVC size of {pvc_size} GB, which equals to 0.75 of the memory sum of "
+            f"all the OSD pods. Then, starting IOs on it"
+        )
+        pvc_obj = pvc_factory(
+            interface=constant.CEPHBLOCKPOOL, size=pvc_size,
+            volume_mode=constants.VOLUME_MODE_BLOCK
+        )
+        pvc_objs.append(pvc_obj)
+        pod_obj = pod_factory(
+            pvc=pvc_obj, pod_dict_path=constants.CSI_RBD_RAW_BLOCK_POD_YAML,
+            raw_block_pv=True
+        )
+        pod_objs.append(pod_obj)
+
+        io_file_size = f"{pvc_size-1}G"
+
+        pod_obj.run_io(
+            storage_type='block', size=io_file_size,
+            runtime=io_run_time, rate='200M', bs='128K', rw_ratio=25
+        )
+
+        pvc_size = int(get_osd_pods_memory_sum() * 0.25)
+        pvc_obj = pvc_factory(
+            interface=constant.CEPHBLOCKPOOL, size=pvc_size,
+            volume_mode=constants.VOLUME_MODE_BLOCK
+        )
+        pvc_objs.append(pvc_obj)
+        pod_obj = pod_factory(
+            pvc=pvc_obj, pod_dict_path=constants.CSI_RBD_RAW_BLOCK_POD_YAML,
+            raw_block_pv=True
+        )
+        pod_objs.append(pod_obj)
+
+        current_throughput = self.cl_obj.calc_average_throughput()
+
+        if target_throughput:
+            if current_throughput > target_throughput * 0.8:
+                self.logger.info(
+                    f"Cluster limit has been reached. It is {current_throughput} Mb/s"
+                )
+                limit_reached = True
+
+        time_to_wait = 60 * 30
+        time_before = time.time()
+        if not target_throughput:
+            self.logger.info(
+                f"\n=====================================================================\n"
+                f"Determining the cluster throughput limit. Once determined, IOs will be"
+                f"\nreduced to load at {target_percentage * 100}% of the cluster limit"
+                f"\n====================================================================="
+            )
+
+        while not limit_reached:
+
+            self.logger.info(
+                f"The cluster average collected throughput BEFORE starting "
+                f"IOs on the newly created pod is {current_throughput} Mb/s"
+            )
+
+            io_file_size = f"{pvc_size - 1}G"
+            pod_obj.run_io(
+                storage_type='block', size=io_file_size,
+                runtime=io_run_time, rate='200M', bs='128K', rw_ratio=25
+            )
+
+            self.logger.info(
+                f"While IO kicks-in on the previously created pod ({pod_obj.name}), "
+                f"creating a new pod and PVC for the next iteration"
+            )
+            pvc_size = int(get_osd_pods_memory_sum() * 0.25)
+            pvc_obj = pvc_factory(
```

Comment:
Not blocking for thie one but It would be nice if we can restrict to < 10 pvc/pods and adjust bs to be mixed config with 265k,1m,4m + rate options to achieve desired throughput, probably we can achieve sooner.

Reply:
Thanks for the suggestions!
- Num of pods is restricted (line 221 - `if len(pod_objs) > 8:)` in case the throughput is lower than 20Mb/s after 8 pods and PVCs have been created and IO was running on them. 
Also, timeout of 15 minutes is set below in case throughput didn't reach more than 20Mb/s.
And, a timeout of 30 in case throughput limit was not determined.

Regarding BS and rate, I set these after experimenting with different values. I can try 256K as BS